src/main/java/Parser.java

```diff
@@ -0,0 +1,63 @@
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Parser {
+    public Parser(Ui ui) {
+    }
+
+    public static void scan(TaskList taskList, Storage storage ) {
+        Scanner sc = new Scanner(System.in);
+        while (true) {
+            String input = sc.nextLine();
+            try {
+                if (input.equals("bye")) {
+                    Ui.printBye();
+                    break;
+                } else if (input.equals("list")) {
+                    Ui.printList(taskList);
+                } else if (input.contains("done")) {
+                    String[] inputArr = input.split(" ");
+                    int taskNum = Integer.parseInt(inputArr[1]);
+                    taskList.taskDone(taskNum);
+                    storage.updateFile(taskList);
+                } else if (input.contains("todo")) {
+                    if (input.split(" ").length == 1) {
+                        throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+                    } else {
+                        taskList.addList(new ToDo(input.substring(5)));
+                        storage.updateFile(taskList);
+                    }
+                } else if (input.contains("deadline")) {
+                    if (input.split(" ").length == 1) {
+                        throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+                    } else {
+                        int index = input.indexOf("/");
+                        taskList.addList(new Deadline(input.substring(9, index), input.substring(index + 4)));
+                    }
+                } else if (input.contains("event")) {
+                    if (input.split(" ").length == 1) {
+                        throw new DukeException("OOPS!!! The description of an event cannot be empty.");
+                    } else {
+                        int index = input.indexOf("/");
+                        taskList.addList(new Event(input.substring(6, index), input.substring(index + 4)));
+                    }
```

Comment:
Maybe it will be better to add in more types of error messges, e.g. command is "event" but there is no venue specified.

Reply:
I agree with the above comment, also, you might be able to consider how to make your codes more generalizable. Now, all codes are written for specific situations, inputs, etc. such as line 29 - length == 1, line 41 - indexOf("/"), line 42 - substring 6. Please think about when you look back at your codes one year later. Can you still remember what the "6" at line 42 means? :)